ntbs-integration-tests/ServiceDirectory/ServiceDirectorySearchTest.cs

```diff
@@ -198,30 +207,43 @@ private static int GetTotalNumberOfResults(IHtmlDocument firstPageContent)
                     .Split(" ", StringSplitOptions.RemoveEmptyEntries)[1]
             );
         }
-
-        private static int GetNumberOfDisplayedItems(IHtmlDocument pageContent)
+        
+        private static int GetNumberOfDisplayedResults(IHtmlDocument pageContent)
         {
             var regionResults = pageContent.GetElementsByClassName("directory-region-search-result-item");
             var serviceResults = pageContent.GetElementsByClassName("directory-service-search-result-item");
             var hospitalResults = pageContent.GetElementsByClassName("directory-hospital-search-result-item");
             var userResults = pageContent.GetElementsByClassName("directory-user-search-result-item");
             return regionResults.Length + serviceResults.Length + hospitalResults.Length + userResults.Length;
         }
-
+        
         private static (
-            IEnumerable<string> secondPageRegionResults,
-            IEnumerable<string> secondPageServiceResults,
-            IEnumerable<string> secondPageHospitalResults,
-            IEnumerable<string> secondPageUserResults)
-            GetSearchResultsTextContents(
-                IHtmlDocument secondPageContent)
+            IEnumerable<string> regionResults,
+            IEnumerable<string> serviceResults,
+            IEnumerable<string> hospitalResults,
+            IEnumerable<string> userResults)
+            GetDisplayedResultsTextContent(IHtmlDocument pageContent) 
         {
-            var secondPageRegionResults = secondPageContent.GetElementsByClassName("directory-region-search-result-item").Select(e => e.TextContent);
-            var secondPageServiceResults = secondPageContent.GetElementsByClassName("directory-service-search-result-item").Select(e => e.TextContent);
-            var secondPageHospitalResults = secondPageContent.GetElementsByClassName("directory-hospital-search-result-item").Select(e => e.TextContent);
-            var secondPageUserResults = secondPageContent.GetElementsByClassName("directory-user-search-result-item").Select(e => e.TextContent);
-            return (secondPageRegionResults, secondPageServiceResults, secondPageHospitalResults,
-                secondPageUserResults);
+            var regionResults = pageContent.GetElementsByClassName("directory-region-search-result-item").Select(e => e.TextContent);
+            var serviceResults = pageContent.GetElementsByClassName("directory-service-search-result-item").Select(e => e.TextContent);
+            var hospitalResults = pageContent.GetElementsByClassName("directory-hospital-search-result-item").Select(e => e.TextContent);
+            var userResults = pageContent.GetElementsByClassName("directory-user-search-result-item").Select(e => e.TextContent);
+            return (regionResults, serviceResults, hospitalResults, userResults);
+        }
+
+        private async Task<int> GetTotalExpectedResults(string searchKeyword)
```

Comment:
I ended up making this change, mostly because it gave me an opportunity to test that `EF.Functions.Like(h.Name, $"%{keyword}%")` was fetching the same results as a .contains on the context